server/test/unit/common/config/test_external_config.py

```diff
@@ -0,0 +1,107 @@
+import os
+import tempfile
+import unittest
+from unittest.mock import patch
+
+import requests
+
+from server.common.config.app_config import ExternalConfig
+from server.common.config.app_config import AppConfig
+from server.test import test_server, FIXTURES_ROOT
+
+
+class TestExternalConfig(unittest.TestCase):
+    def test_type_convert(self):
+        # The values from environment variables and aws secrets are returned as strings.
+        # These values need to be converted to the proper types.
+
+        self.assertEqual(ExternalConfig.convert_value("1"), int(1))
+        self.assertEqual(ExternalConfig.convert_value("1.1"), float(1.1))
+        self.assertEqual(ExternalConfig.convert_value("string"), "string")
+        self.assertEqual(ExternalConfig.convert_value("true"), True)
+        self.assertEqual(ExternalConfig.convert_value("True"), True)
+        self.assertEqual(ExternalConfig.convert_value("false"), False)
+        self.assertEqual(ExternalConfig.convert_value("False"), False)
+        self.assertEqual(ExternalConfig.convert_value("False"), False)
+        self.assertEqual(ExternalConfig.convert_value("null"), None)
+        self.assertEqual(ExternalConfig.convert_value("None"), None)
+        self.assertEqual(ExternalConfig.convert_value("{'a':10, 'b':'string'}"), dict(a=int(10), b="string"))
+
+    def test_environment_variable(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            configfile = os.path.join(tempdir, "config.yaml")
```

Comment:
Could you create a helper method for creating an external config for testing and add it to the others in config/__init__?